assets/shared/components/modal/index.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * WordPress dependencies
+ */
+import { Icon, close as closeIcon } from '@wordpress/icons';
+
+/**
+ * Modal component.
+ *
+ * @param {Object}   props
+ * @param {boolean}  props.isOpen      Whether the modal should be open or not.
+ * @param {Function} props.handleClose Callback to run when trying to close the modal.
```

Comment:
WDYT about renaming this prop to `onClose`? 

It makes more sense considering that it's only an event handler in the current state. :)